test-suite/Makefile

```diff
@@ -242,6 +246,45 @@ $(addsuffix .log,$(wildcard bugs/closed/*.v)): %.v.log: %.v
 	  fi; \
 	} > "$@"
 
+#######################################################################
+# Unit tests
+#######################################################################
+
+UNITTEST_OCAMLOPT=ocamlfind opt
+UNITTEST_CAMLFLAGS=-rectypes
+UNITTEST_COQDIRS=-I $(LIB)/clib -I $(LIB)/lib -I $(LIB)/kernel -I $(LIB)/kernel/byterun -I $(LIB)/intf \
+-I $(LIB)/kernel/byterun -I $(LIB)/library -I $(LIB)/engine -I $(LIB)/pretyping -I $(LIB)/interp \
+-I $(LIB)/proofs -I $(LIB)/parsing -I $(LIB)/printing -I $(LIB)/tactics -I $(LIB)/vernac \
+-I $(LIB)/stm -I $(LIB)/toplevel
+
+# ML files from unit-test framework, not containing tests
+UNITTEST_SRCFILES=$(shell find ./unit-tests/src -name *.ml)
+UNITTEST_ALLMLFILES:=$(shell find ./unit-tests -name *.ml)
+UNITTEST_MLFILES:=$(filter-out $(UNITTEST_SRCFILES),$(UNITTEST_ALLMLFILES))
+UNITTEST_LOGFILES:=$(patsubst %.ml,%.ml.log,$(UNITTEST_MLFILES))
+
+UNITTEST_CMXS=utest.cmx
+
+# this is same link order as CORECMA in Makefile.common in Coq root dir
+UNITTEST_COQCMXAS=clib.cmxa lib.cmxa kernel.cmxa library.cmxa engine.cmxa pretyping.cmxa \
+ interp.cmxa proofs.cmxa parsing.cmxa printing.cmxa tactics.cmxa vernac.cmxa stm.cmxa toplevel.cmxa
```

Comment:
Given that we use ocamlfind, we should use `META.coq` to link these `cmxa`, not hard wire them here.

Reply:
How does using META work?

Reply:
How about `include ../Makefile.common`, then referring to LINKCMX?

And if not, shouldn't the line be more like `UNIT_TEST_COQCMXAS=$(OCAMLFIND) opt -package coq.toplevel ...`?  What I should use for `usual-oamlc-arguments`, I assume that was a placeholder for something else.  I couldn't figure it out from the existing Makefiles.

The find doesn't work from the command line:
```
$ ocamlfind opt -package coq.toplevel
ocamlfind: Package `coq.toplevel' not found
```

Reply:
This is because you did not add to OCAMLPATH the directory containing `META.coq`

It depends what you want to do. Using the `META` is what a client of Coq would do (IMHO the test suite could be seen as client, this is also what the comment you quoted was saying). Using `Makefile.common` is what a component of Coq would do.

I'm fine with any solution not duplicating info (such as the path of ocamlfind and the list of cmxa files).
As I said running the test suite against an installed Coq is not a useful use case, so reusing `Makefile.common` is OK, if you find this solution simpler.